### PR TITLE
Fix undefined object errors when loading assets in workers

### PR DESCRIPTION
**Pull request by jbaicoianu**
Loading model assets inside of workers currently throws errors trying to access undefined objects, like ```window``` and ```HTMLCanvasElement```.  This pull request fixes those errors.

### CODE REVIEW

File: src/extras/ImageUtils.js [hunk 1]
```diff
@@ -10,7 +10,11 @@ var ImageUtils = {
 
 		var canvas;
 
-		if ( image instanceof HTMLCanvasElement ) {
+		if ( typeof HTMLCanvasElement == 'undefined' ) {
+
+			return image.src;
+
+		} if ( image instanceof HTMLCanvasElement ) {
```
Comment (WestLangley) [inline src/extras/ImageUtils.js]: either `else if` or fix formating

**Comment by takahirox** [comment]
I'm trying to fix the same issue in other modules like glTF loader. One thing coming to my mind. Should we avoid using `window` as much as possible to let Three.js be friendly to Worker? If so, feeling like we need to update the code style.

**Comment by jbaicoianu** [comment]
> Should we avoid using `window` as much as possible to let Three.js be friendly to Worker? If so, feeling like we need to update the code style.

My understanding is that there's no benefit to explicitly using window/self when referring to globally-available functions unless you have a need to call using a variable function name with syntax like ```lwindow[functionName](functionArg)```, or if you want to be explicit about your scope when overriding functions with similar names in different scopes, which I don't think Three.js does anywhere.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**